src/commands/build/wranglerjs/output.rs

```diff
@@ -20,33 +29,72 @@ pub struct WranglerjsOutput {
 }
 
 impl WranglerjsOutput {
+    pub fn to_worker_bundle(&self) -> Result<WorkerBundle, failure::Error> {
+        let dist = env::temp_dir().join("wrangler_webpack");
+
+        let mut bindings = vec![];
+        let script_path = Path::new(&dist).join("script.js".to_string());
+
+        let bundle_path = Path::new(&dist);
+        if !bundle_path.exists() {
+            fs::create_dir(bundle_path)?;
+        }
+
+        let mut script_file = File::create(&script_path)?;
+        let mut script = create_prologue();
+        script += &self.script;
+
+        let wasm_path = if let Some(encoded_wasm) = &self.wasm {
+            let path = Path::new(&dist).join("module.wasm".to_string());
+
+            let wasm = decode(encoded_wasm).expect("could not decode Wasm in base64");
+            let mut wasm_file = File::create(&path)?;
+            wasm_file.write_all(&wasm)?;
+
+            bindings.push(Binding::new_wasm_module(
+                WASM_BINDING.to_string(), // name
+                WASM_BINDING.to_string(), // part
+            ));
+
+            Some(path)
+        } else {
+            None
+        };
+
+        script_file.write_all(script.as_bytes())?;
+
+        // cleanup {Webpack} dist, if specified.
+        if let Some(dist_to_clean) = &self.dist_to_clean {
+            info!("Remove {}", dist_to_clean);
+            fs::remove_dir_all(dist_to_clean).expect("could not clean Webpack dist.");
+        }
+
+        fs::remove_dir_all(dist).expect("could not clean temp dist.");
+
+        Ok(WorkerBundle {
+            script_path,
+            wasm_path,
+            wasm_binding: Some(WASM_BINDING.to_string()),
+            bindings,
+        })
+    }
+
     pub fn has_errors(&self) -> bool {
         !self.errors.is_empty()
     }
 
     pub fn get_errors(&self) -> String {
         self.errors.join("\n")
     }
+}
 
-    pub fn script_size(&self) -> String {
-        let mut e = ZlibEncoder::new(Vec::new(), Compression::default());
-        e.write_all(&self.script.as_bytes())
-            .expect("could not write buffer");
-        let compressed_bytes = e.finish();
-
-        match NumberPrefix::decimal(compressed_bytes.unwrap().len() as f64) {
-            Standalone(bytes) => format!("{} bytes", bytes),
-            Prefixed(prefix, n) => format!("{:.0} {}B", n, prefix),
-        }
-    }
-
-    pub fn wasm_size(&self) -> String {
-        let size = self.wasm.to_owned().unwrap().len();
-        match NumberPrefix::decimal(size as f64) {
-            Standalone(bytes) => format!("{} bytes", bytes),
-            Prefixed(prefix, n) => format!("{:.0} {}B", n, prefix),
-        }
-    }
+// We inject some code at the top-level of the Worker; called {prologue}.
+// This aims to provide additional support, for instance providing {window}.
+pub fn create_prologue() -> String {
```

Comment:
This seems out of scope for this change.

Reply:
I just moved the code from the bundle.rs file, that I deleted in favor of the shared WorkerBundle.